server.js

```diff
@@ -1,39 +1,112 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
-
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
-
-// Start defining your routes here
+import express from 'express';
+import cors from 'cors';
+import mongoose from 'mongoose';
+import dotenv from 'dotenv';
+
+import booksData from './data/books.json';
+
+dotenv.config();
+
+const mongoUrl = process.env.MONGO_URL || 'mongodb://localhost/project-mongo';
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+const bookSchema = new mongoose.Schema({
+  title: String,
+  authors: String,
+  average_rating: Number,
+  language_code: String,
+  num_pages: Number,
+  isbn13: Number,
+});
+
+const Book = mongoose.model('Book', bookSchema);
+
+if (process.env.RESET_DB) {
+  const seedDB = async () => {
+    await Book.deleteMany();
+
+    booksData.forEach((item) => {
+      new Book(item).save();
+    });
+  };
+  seedDB();
+}
+const port = process.env.PORT || 8080;
+const app = express();
+
+app.use(cors());
+app.use(express.json());
+
+// Routes
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send('Welcome to the Book API');
+});
+
+// Query parameter to get all the books unless a title or author was specified
+app.get('/books', async (req, res) => {
+  const { title, authors } = req.query;
+
+  const books = await Book.aggregate([
+    {
+      $match: {
+        title: {
+          $regex: new RegExp(title || '', 'i')
+        },
+        authors: {
+          $regex: new RegExp(authors || '', 'i')
+        }
+      }
+    }
+    // {
+    //   $sort: {
+    //     average_rating: +sort,
+    //   },
+    // }
+    // {
+    //   $skip: Number((page - 1) * per_page + 1)
+    // },
+    // {
+    //   $limit: Number(per_page)
```

Comment:
you need to provide the default per_page value because here you trying parsing undefined into the number if your user will not provide you with value in the query.

When something is yealing about value, remember to console.log value the input before and after the function.
``` js
console.log(typeof(per_page))
const numberPefPage = Number(per_page)
console.log(typeof(numberPerPage))
```
and then you would see

```
< undefined
< NaN
```